[Docs] document that `.child()` call triggers `level-change` event. (#1014)
pinojs/pino: docs/api.md

logger.level = 'trace' // trigger event
```

> Please note that due to a [known bug](https://github.com/pinojs/pino/issues/1006), every `logger.child()` call will
> fire a `level-change` event. You might want to ignore these events, using something like:

Reviewer: ```suggestion
Please note that due to a [known bug](https://github.com/pinojs/pino/issues/1006), every `logger.child()` call will
fire a `level-change` event. These events can be ignored by writing an event handler like:
```

You might want to ignore these events, using something like:

```js
const logger = require('pino')()

> logger.on('level-change', (lvl, val, prevLvl, prevVal) => {

Reviewer 2: I would remove the arrow function and rely on `this === logger` to check.

Author: Thank you, I did not think of this (no pun intended) :)

>   if (logger === this) {
>     console.log('%s (%d) was changed to %s (%d)', prevLvl, prevVal, lvl, val)
>   }

Reviewer: ```suggestion
  if (logger !== this) {
    return
  }
  console.log('%s (%d) was changed to %s (%d)', prevLvl, prevVal, lvl, val)
```
